File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = prices[0];
        int res = 0;
        for(int i = 1; i < prices.size(); i++)
        {
            if(prices[i] < minPrice)
                minPrice = prices[i];
            int profit = prices[i] - minPrice;
            res = max(profit, res);
        }
        return res;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = float('inf')
        res = 0
        for x in prices:
            if x < minPrice:
                minPrice = x
            else:
                profit = x - minPrice
                res = max(res, profit)
        return res